ArduPlane/AP_ExternalControl_Plane.cpp

```diff
@@ -19,4 +19,37 @@ bool AP_ExternalControl_Plane::set_global_position(const Location& loc)
     return plane.set_target_location(loc);
 }
 
+/*
+  Sets the target airspeed.
+*/
+bool AP_ExternalControl_Plane::set_airspeed(const float airspeed)
+{
+#if AP_PLANE_OFFBOARD_GUIDED_SLEW_ENABLED
+    // The command is only valid in guided mode.
+    if (plane.control_mode != &plane.mode_guided) {
+        return false;
+    }
+
+    // Assume the user wanted maximum acceleration.
+    const float acc_instant = 0.0;
+    return plane.mode_guided.handle_change_airspeed(airspeed, acc_instant);
+#else 
+  return false;
+#endif
+}
+
+/*
+    Request gliding
+*/
+bool AP_ExternalControl_Plane::request_gliding(bool gliding_requested)
+{
+    // The command is only valid in guided mode.
+    if (plane.control_mode != &plane.mode_guided) {
+        return false;
+    }
+
+    plane.TECS_controller.set_gliding_requested_flag(gliding_requested);
```

Comment:
Does this get reverted somewhere on leaving mode-guided?  You mentioned doing in it mode_guided on Discord.

Reply:
I have added that now (unconditionally). I don't see any reason we would persist that flag over mode switch, so it seems like a safe addition to ModeGuided.

Reply:
Likewise what happens if we switch to a type of guided where TECS is not used?

Reply:
That would potentially be bad. Do you have any recommendations on how to detect that given there isn't a clear interface in ModeGuided to know if TECS is being used?